src/solve.jl

```diff
@@ -169,6 +169,47 @@ function init(prob::AbstractDDEProblem{uType,tType,lType,isinplace}, alg::algTyp
         save_idxs = nothing # prevents indexing of ODE solution and saves memory
     end
 
+    # new cache required for methods that wrap derivatives of f such as Rosenbrock
+    # create uprev2 in same way as in OrdinaryDiffEq
+    if integrator.uprev === integrator.uprev2
+        uprev2 = uprev
+    else
+        if uType <: Array
+            uprev2 = copy(uprev)
+        else
+            uprev2 = deepcopy(uprev)
+        end
+    end
+    dde_cache = OrdinaryDiffEq.alg_cache(alg.alg, u, integrator.rate_prototype,
```

Comment:
This is going to double the allocations though. Is there no better way? Why can't the derivative wrapper use the `f` from the delay equation, just differently?

Reply:
At the point when the cache and hence also the derivative wrappers are created the `f` with 3 arguments that implicitly contains the history function is not existent yet. If we just pass `f` (as currently done) then we somehow would also have to provide the history function every time the wrappers are called. Is there an easy way to do that? I didn't know how to solve this problem easily in another way.